books/serializers.py

```diff
@@ -0,0 +1,53 @@
+from rest_framework import serializers
+
+from books.models.books import Book, Author, Publisher
+from books.models.country import Country
+
+
+class AuthorSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(read_only=True)
+
+    class Meta:
+        model = Author
+        fields = ('id', 'name')
+
+
+class PublisherSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(read_only=True)
+
+    class Meta:
+        model = Publisher
+        fields = ('id', 'name')
+
+
+class CountrySerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(read_only=True)
+
+    class Meta:
+        model = Country
+        fields = ('id', 'name')
+
+
+class BookSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(read_only=True)
+
+    class Meta:
+        model = Book
+        fields = ("id", "name", "isbn", "authors", "number_of_pages", "publisher", "country", "release_date")
+
+    def to_representation(self, instance):
+        data = super(BookSerializer, self).to_representation(instance)
+        data['country'] = CountrySerializer(instance.country).data
+        data['publisher'] = PublisherSerializer(instance.publisher).data
+        data['authors'] = AuthorSerializer(instance.authors, many=True).data
+        return data
```

Comment:
@xhahid43eb Still no need of overriding to_representation, you can do something like this.
```suggestion
class BookSerializer(serializers.ModelSerializer):
    id = serializers.IntegerField(read_only=True)
    country = CountrySerializer(source='country')
    publisher = PublisherSerializer(source='publisher')
    authors = AuthorSerializer(source='authors_set', many=True)

    class Meta:
        model = Book
        fields = ("id", "name", "isbn", "authors", "number_of_pages", "publisher", "country", "release_date")
```